modules/ocf_hpc/files/add_slurm_users.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+'''
+Script that adds users in the 'ocfhpc' LDAP group to SLURM as a SLURM user.
+'''
+import subprocess
+
+from ocflib.account.utils import list_group
+
+
+DEFAULT_ACCOUNT = 'users'
+
+
+def main():
+    ocfhpc_ldap_user_set = set(list_group('ocfhpc'))
+
+    # This produces a newline-separated list of SLURM users.
+    ocfhpc_slurm_user_list = subprocess.run(
+        ['sacctmgr', '-noP', 'list', 'users', 'format=User'],
+        stdout=subprocess.PIPE,
+        check=True
+    ).stdout.decode().split('\n')[:-1]
+    ocfhpc_slurm_user_set = set(ocfhpc_slurm_user_list)
+
+    users_to_be_added = ocfhpc_ldap_user_set.difference(ocfhpc_slurm_user_set)
+
+    if not users_to_be_added:
+        return
+
+    for new_user in users_to_be_added:
+        # Add a new user to SLURM and commit changes immediately.
+        subprocess.run(['sacctmgr', '-i', 'add', 'user', new_user, 'account={}'.format(DEFAULT_ACCOUNT)])
```

Comment:
I would double check (if you haven't already) that this command adds users correctly, because in the past I believe I've had to specify the cluster and/or partition.

Reply:
Yep, it does. Default cluster is ocf-hpc, and apparently it doesn't need a partition.